src/components/Logout/LogoutScreen.test.tsx

```diff
@@ -0,0 +1,298 @@
+};
+
+describe("LogoutScreen", () => {
+  beforeEach(() => {
+    mockCaptureException.mockReset().mockName("captureException");
+    mockNavigate.mockReset().mockName("navigate");
+    mockCallLogout.mockReset().mockName("callLogout");
+  });
+
+  it("should attempt logout from all campaigns", async () => {
+    expect.assertions(5);
+
+    render(
+      <ConfigContext.Provider value={defaultConfigContextProviderValue}>
+        <AuthStoreContext.Provider value={defaultAuthStoreContextValue}>
+          <CampaignConfigsStoreContext.Provider
+            value={defaultCampaignConfigStoreContextValue}
+          >
+            <ImportantMessageSetterContext.Provider value={jest.fn()}>
+              <LogoutScreenContainer />
+            </ImportantMessageSetterContext.Provider>
+          </CampaignConfigsStoreContext.Provider>
+        </AuthStoreContext.Provider>
+      </ConfigContext.Provider>
+    );
+
+    await waitFor(() => expect(mockNavigate).toHaveBeenCalledTimes(1));
+
+    expect(mockCallLogout).toHaveBeenCalledTimes(3);
+    expect(mockCallLogout).toHaveBeenCalledWith("sesA", "optA", "endA");
+    expect(mockCallLogout).toHaveBeenCalledWith("sesB", "optB", "endB");
+    expect(mockCallLogout).toHaveBeenCalledWith("sesC", "optC", "endC");
+  });
+
+  it("should logout from campaigns with no error and vice versa", async () => {
+    expect.assertions(9);
+
+    mockCallLogout.mockImplementation(async (sessionToken) => {
+      if (sessionToken === "sesA") throw new LogoutError("");
+    });
+    const mockRemoveAuthCredentials = jest
+      .fn()
+      .mockName("removeAuthCredentials");
+    const mockRemoveCampaignConfig = jest.fn().mockName("removeCampaignConfig");
+
+    render(
+      <AuthStoreContext.Provider
+        value={{
+          ...defaultAuthStoreContextValue,
+          removeAuthCredentials: mockRemoveAuthCredentials,
+        }}
+      >
+        <CampaignConfigsStoreContext.Provider
+          value={{
+            ...defaultCampaignConfigStoreContextValue,
+            removeCampaignConfig: mockRemoveCampaignConfig,
+          }}
+        >
+          <LogoutScreenContainer />
+        </CampaignConfigsStoreContext.Provider>
+      </AuthStoreContext.Provider>
+    );
+
+    await waitFor(() => expect(mockNavigate).toHaveBeenCalledTimes(1));
+
+    expect(mockRemoveAuthCredentials).toHaveBeenCalledTimes(2);
+    expect(mockRemoveAuthCredentials).not.toHaveBeenCalledWith(keyA);
+    expect(mockRemoveAuthCredentials).toHaveBeenCalledWith(keyB);
+    expect(mockRemoveAuthCredentials).toHaveBeenCalledWith(keyC);
+
+    expect(mockRemoveCampaignConfig).toHaveBeenCalledTimes(2);
+    expect(mockRemoveCampaignConfig).not.toHaveBeenCalledWith(keyA);
+    expect(mockRemoveCampaignConfig).toHaveBeenCalledWith(keyB);
+    expect(mockRemoveCampaignConfig).toHaveBeenCalledWith(keyC);
+  });
+
+  it("should clear all credentials and configs and move to login screen upon successful logout", async () => {
```

Comment:
Can this be done under the first `it` above?

Reply:
i refactored it slightly differently to sort of match the structure you gave [here](https://github.com/rationally-app/mobile-application/pull/416#discussion_r694030520)